File: app/src/main/java/com/example/bloodbank/ui/fragment/BaseFragment.java
package com.example.bloodbank.ui.fragment;

import androidx.fragment.app.Fragment;

import com.example.bloodbank.ui.activity.BaseActivity;
import com.example.bloodbank.ui.activity.HomeActivity;

public class BaseFragment extends Fragment {
    public BaseActivity baseActivity;
    HomeActivity homeActivity;

    public void setupAcvtivity() {
        baseActivity = (BaseActivity) getActivity();
        baseActivity.baseFragment = this;
    }

    public void onBack() {
        baseActivity.superBackPressed();
    }
}

File: app/src/main/java/com/example/bloodbank/ui/fragment/BaseFragment.kt
package com.example.bloodbank.ui.fragment

import androidx.fragment.app.Fragment
import com.example.bloodbank.ui.activity.BaseActivity
import com.example.bloodbank.ui.activity.HomeActivity

open class BaseFragment : Fragment() {
    @JvmField
    var baseActivity: BaseActivity? = null
    var homeActivity: HomeActivity? = null
    fun setupActivity() {
        baseActivity = activity as BaseActivity?
        baseActivity!!.baseFragment = this
    }

    open fun onBack() {
        baseActivity!!.superBackPressed()
    }
}